src/celrender/cometrenderer.h

```diff
@@ -0,0 +1,64 @@
+// cometrenderer.h
+//
+// Copyright (C) 2001-present, the Celestia Development Team
+// Original version by Chris Laurel <claurel@gmail.com>
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+
+#include <utility>
+#include <Eigen/Core>
+
+class Body;
+class Observer;
+class Renderer;
+class CelestiaGLProgram;
+struct Matrices;
+
+namespace celestia::render
+{
+class IndexedVertexObject;
+
+class CometRenderer
+{
+public:
+    explicit CometRenderer(Renderer &renderer);
+    ~CometRenderer() = default;
+    CometRenderer(const CometRenderer&) = delete;
+    CometRenderer(CometRenderer&&) = default;
```

Comment:
Clang 15.0.2 gives a warning because the reference type member implicitly deletes the move constructor and move assignment operators (similarly for `AtmosphereRenderer`)

Reply:
let's remove them explicitly as we don't use them now and if we need them sometime then implement properly.